packages/nimble-components/src/table/specs/table-column-specs/table-column-mapping.md

```diff
@@ -0,0 +1,359 @@
+
+_Content_
+
+-   column title (text or icon)
+-   1 or more `nimble-mapping-icon` or `nimble-mapping-spinner` elements
+
+#### General mapping column element:
+
+_Component Name_
+
+-   `nimble-table-column-mapping`
+
+_Props/Attrs_
+
+-   `field-name`: string
+-   `key-type`: 'string' | 'number' | 'boolean'
+-   `fractional-width`: number (defaults to 1)
+-   `min-pixel-width`: number (defaults to minimum supported by table)
+
+_Content_
+
+-   column title (text or icon)
+-   1 or more `nimble-mapping-text` elements
+
+#### Mapping element (icon):
+
+_Component Name_
+
+-   `nimble-mapping-icon`
+
+_Props/Attrs_
+
+-   `key`: string | number | boolean | null
+-   `icon`: string - name of the Nimble icon element
+-   `severity`: string - one of the supported enum values. Controls color of the icon.
+-   `label`: string - localized value used as the accessible name and `title` of the icon. Will also be displayed in the group header.
+-   `default-mapping`: boolean - presence causes this mapping to be used when no others match the value
+
+#### Mapping element (spinner):
+
+_Component Name_
+
+-   `nimble-mapping-spinner`
+
+_Props/Attrs_
+
+-   `key`: string | number | boolean | null
+-   `label`: string - localized value used as the accessible name and `title` of the spinner. Will also be displayed in the group header.
+-   `default-mapping`: boolean - presence causes this mapping to be used when no others match the value
+
+#### Mapping element (text):
+
+_Component Name_
+
+-   `nimble-mapping-text`
+
+_Props/Attrs_
+
+-   `key`: string | number | boolean | null
+-   `label`: string - display text
+-   `default-mapping`: boolean - presence causes this mapping to be used when no others match the value
+
+### Anatomy
+
+#### `nimble-table-column-mapping`
+
+```HTML
+<template slot="${x => x.columnInternals.uniqueId}">
+    <slot
+        ${slotted('mappings')}
+        name="mapping">
+    </slot>
+    <span class="header-content">
+        <slot></slot>
+    </span>
+</template>
+```
+
+Cell view:
+
+The cell view relies on the matched mapping to provide a template to render.
+
+```HTML
+<div ${ref('wrapper') class="wrapper"}>
+    ${repeat(x => x.columnConfig.mappings,
+        html<Mapping, TableColumnMappingCellView>`
+            ${when((x, c) => x.key === (c.parent as TableColumnMappingCellView).cellRecord.value,
+                html<Mapping>`${x => x.cellViewTemplate}`
+            )}
+        `
+    )}
+</div>
+${when(x => x.wrapper.childElementCount === 0, html<TableColumnMappingCellView>`
+    ${repeat(x => x.columnConfig.mappings,
+        html<Mapping, TableColumnMappingCellView>`
+            ${when((x, c) => x.defaultMapping,
+                html<Mapping>`${x => x.cellViewTemplate}`
+            )}
+        `
+    )}
+`)}
+```
+
+Group header view:
+
+Similarly, the group header view relies on the matched mapping to provide a template to render.
+
+```HTML
```

Comment:
Same here. It seems like part of this conditional logic should really belongs in the group header view class, not the template.